demo/long_video_demo.py

```diff
@@ -120,38 +130,37 @@ def show_results():
     cv2.destroyAllWindows()
 
 
-def inference():
-    if len(frame_queue) != sample_length:
+def inference(model, data, args, frame_queue):
+    if len(frame_queue) != args.sample_length:
         # Do no inference when there is no enough frames
         return False, None
 
     cur_windows = list(np.array(frame_queue))
-    img = frame_queue.popleft()
     if data['img_shape'] is None:
-        data['img_shape'] = img.shape[:2]
+        data['img_shape'] = frame_queue[0].shape[:2]
+
     cur_data = data.copy()
     cur_data['imgs'] = cur_windows
-    cur_data = test_pipeline(cur_data)
+    cur_data = args.test_pipeline(cur_data)
     cur_data = collate([cur_data], samples_per_gpu=1)
     if next(model.parameters()).is_cuda:
-        cur_data = scatter(cur_data, [device])[0]
+        cur_data = scatter(cur_data, [args.device])[0]
     with torch.no_grad():
         scores = model(return_loss=False, **cur_data)[0]
+
+    if args.stride > 0:
+        pred_stride = int(args.sample_length * args.stride)
+        for i in range(pred_stride):
+            frame_queue.popleft()
```

Comment:
it seems not right. If `args.stride = 0` (not assigned),   `frame_queue` will not be popped ?

Reply:
OK, if the `deque` is full, it will automatically pop : )